docs/remote-devcontainer.md

```diff
@@ -0,0 +1,47 @@
+# Contributing with VS Code Remote Container Extension
+
+ We have laid out steps for contributing to the **Azure Key Vault Provider for Secrets Store CSI Driver** using the `VS Code - Remote Container Extension`.
+
+## Prerequisites
+1. Azure Subscription
+
+## Fork and Clone Repository
+
+Before we dive into setting up a remote container environment, fork and clone the repository first. Once cloned, enter into the `root` folder of the project:
+
+  ```bash
+  $ cd secrets-store-csi-driver-provider-azure
+  ```
+
+## VS Code with Remote Container Extension
+
+The [VS Code Remote Container Extension](https://code.visualstudio.com/docs/remote/containers) utilizes the `.devcontainer` folder to build a remote container that will have all necessary dependencies installed to contribute to the **Azure Key Vault Provider for Secrets Store CSI Driver**.
+
+### Dependencies Included Inside The Remote Container
+
+- `yq and jq` command line utilities for manipulating YAML and JSON files
+- `Azure CLI` for access to your Azure Subscription
+  - Your `.azure` folder on your host machine is mounted into the container, so you will be logged in to the same Azure Subscription.
```

Comment:
Is this true? I see the `.azure` folder mounted but still need to login inside the container -

```
root@docker-desktop:/go/src/secrets-store-csi-driver-provider-azure# az account show -o table
Please run 'az login' to setup account.
```

Reply:
@aramase it seems to not be the case, it won't automatically log you in this way anymore. the `/.azure` folder contains all subscriptions info but the user won't be logged in. Since this is the case, do you think it's better to just notify them to run `az login` to log in to their azure account?

Reply:
@bdlb77 yeah, that makes sense! let's do that.

Reply:
@aramase could you give it another shot with az login? I changed the mount point of the `.azure. folder and it seemed to work for me :)

Reply:
Tried out it and I'm logged in now 👍

Reply:
Where do we tell users to run `az login`?

Reply:
It's not required. They are logged in by default when `.azure` folder is mounted.

Reply:
if they are already logged in to azure, they will continue to be logged in... If they haven't logged in then they are able to do so from within the container